CMakeLists.txt

```diff
@@ -173,6 +173,11 @@ if ("${SPIRV_SKIP_EXECUTABLES}")
   set(SPIRV_SKIP_TESTS ON)
 endif()
 
+option(SPIRV_CHECK_CONTEXT "In a debug build, check if the IR context is in a valid state." ON)
```

Comment:
Why not all the time?

Reply:
I can be an expensive check.  So I would like to allow developers to turn it off if they want.  It is on by default for the debug build.

Reply:
Ah, OK.  Could you document why it's an option.  LGTM with that change.